src/coprocessor/endpoint.rs

```diff
@@ -67,35 +68,55 @@ pub struct Host {
     sched: Scheduler<Task>,
     reqs: HashMap<u64, Vec<RequestTask>>,
     last_req_id: u64,
-    pool: ThreadPool<DummyContext>,
-    low_priority_pool: ThreadPool<DummyContext>,
-    high_priority_pool: ThreadPool<DummyContext>,
+    pool: ThreadPool<CopContext>,
+    low_priority_pool: ThreadPool<CopContext>,
+    high_priority_pool: ThreadPool<CopContext>,
     max_running_task_count: usize,
 }
 
-// TODO: remove these dummy structures.
 #[derive(Clone)]
-struct DummyContext {}
+struct CopContext {
+    raftstore_sender: Option<Sender<Statistics>>,
+    statistics: Statistics,
+}
 
-unsafe impl Send for DummyContext {}
+unsafe impl Send for CopContext {}
 
-impl Context for DummyContext {
+impl Context for CopContext {
     fn on_task_started(&mut self) {}
     fn on_task_finished(&mut self) {}
-    fn on_tick(&mut self) {}
+    fn on_tick(&mut self) {
+        if let Some(ref sender) = self.raftstore_sender {
```

Comment:
the default timeout is  20ms. If we do nothing, we still need to send the metrics.

I think we should enlarge the default timeout for the thread pool, maybe 500ms or 1s.